README.md

```diff
@@ -7,13 +7,120 @@ api-filter
 
 Parser/builder for filters from api query parameters.
 
+It is just a parser/builder for filters, it is not a place for business logic so it should be wrapped by your class, if you want to be more strict about filters.
+Same if you want different settings per entity/table, it should be done by specific wrapper around this library.
+
+
 ## Installation
 ```bash
 composer require lmc/api-filter
 ```
 
 
+## Filters
+
+### EQ (=)
+```http request
+GET http://host/endpoint/?field[eq]=value
+GET http://host/endpoint/?field=value
+```
+_All examples ☝️ are equal_
+
+### GT (>)
+```http request
+GET http://host/endpoint/?field[gt]=value
+```
+
+### GTE (>=)
+```http request
+GET http://host/endpoint/?field[gte]=value
+```
+
+### LT (<)
+```http request
+GET http://host/endpoint/?field[lt]=value
+```
+
+### LTE (<=)
+```http request
+GET http://host/endpoint/?field[lt]=value
+```
+
+### IN
+```http request
+GET http://host/endpoint/?type[in][]=one&type[in][]=two
+```
+_☝️ is not implemented yet_
+
+
 ## Usage
+For example lets have query parameters from following request
+```http request
+GET http://host/endpoint/?field=value
+```
+
+### Initialization
+```php
+// in DI container/factory
+$apiFilter = new ApiFilter();
+$apiFilter->registerApplicator(...);  // optional
+$apiFilter->registerEscape(...);      // optional
+
+// in service/controller/...
+$filters = $apiFiter->parseFilters($request->query->all());
+```
+
+### With basic `SQL Applicator`
+```php
+// in Model/EntityRepository
+$sql = 'SELECT * FROM table';
+$sql = $apiFilter->applyFilters($filters, $sql);    // "SELECT * FROM table WHERE 1 AND field = 'value'"
```

Comment:
`WHERE 1`?

Reply:
It is the easiest way of adding multiple conditions. I can just "implode" conditions by `and` and concat with `sql` by another `and`. 

But in the future, there might be only NON sql applicators, so this "feature" will probably be removed in the future.